File: backend/models/period/period.py
from datetime import datetime
from typing import Optional


class Period:
    def __init__(
        self,
        start_date: datetime,
        end_date: datetime,
        spent_amount: Optional[float] = 0.0,
    ):
        self.start_date = start_date
        self.end_date = end_date
        self._spent_amount = spent_amount

    def to_dict(self) -> dict:
        return {
            "start_date": self.start_date.isoformat(),
            "end_date": self.end_date.isoformat(),
            "spent_amount": self._spent_amount,
        }

File: frontend/src/Models/Period.ts
export interface Period {
    start_date: string;
    end_date: string;
    spent_amount: number; // use number here now
    income_amount: number;
  }
  
  export function parsePeriod(raw: any): Period {
    return {
      start_date: raw.start_date,
      end_date: raw.end_date,
      spent_amount: parseFloat(raw.spent_amount).toFixed(2), // returns string, cast after
      income_amount: parseFloat(raw.income_amount).toFixed(2),
    } as unknown as Period; // cast only if you keep interface with numbers
  }
  